server/src/test/java/org/apache/cassandra/sidecar/concurrent/AsyncConcurrentTaskExecutorTest.java

```diff
@@ -0,0 +1,487 @@
+import static org.assertj.core.api.Assertions.assertThat;
+
+@ExtendWith(VertxExtension.class)
+class AsyncConcurrentTaskExecutorTest
+{
+    private final Vertx vertx = Vertx.vertx();
+
+    @Test
+    public void testTaskFailure(VertxTestContext context)
+    {
+        List<Callable<Future<Boolean>>> tasks = new ArrayList<>();
+        String failureMessage = "Task failed";
+        tasks.add(getDummyFailedTask(failureMessage));
+        AsyncConcurrentTaskExecutor<Boolean> concurrentTaskExecutor = new AsyncConcurrentTaskExecutor<>(vertx, tasks, 1);
+        List<Future<Boolean>> taskFutures = concurrentTaskExecutor.start();
+        Future.join(taskFutures).onComplete(context.failing(result -> context.verify(() -> {
+            assertThat(result).isNotNull();
+            assertThat(result.getMessage()).isEqualTo(failureMessage);
+            assertThat(taskFutures.get(0).failed()).isTrue();
+            context.completeNow();
+        })));
+    }
+
+    @Test
+    void testZeroTasks(VertxTestContext context)
+    {
+        List<Callable<Future<Boolean>>> tasks = new ArrayList<>();
+        AsyncConcurrentTaskExecutor<Boolean> concurrentTaskExecutor = new AsyncConcurrentTaskExecutor<>(vertx, tasks, 1);
+        List<Future<Boolean>> taskFutures = concurrentTaskExecutor.start();
+        Future.join(taskFutures).onComplete(ar -> context.verify(() -> {
+            assertThat(ar).isNotNull();
+            assertThat(ar.result().list().size()).isEqualTo(0);
+            context.completeNow();
+        }));
+    }
+
+    @Test
+    public void testMultipleTaskFailure(VertxTestContext context)
+    {
+        List<Callable<Future<Boolean>>> tasks = new ArrayList<>();
+        for (int i = 0; i < 10; i++)
+        {
+            String failureMessage = "Task " + i + " failed.";
+            tasks.add(getDummyFailedTask(failureMessage));
+        }
+        AsyncConcurrentTaskExecutor<Boolean> concurrentTaskExecutor = new AsyncConcurrentTaskExecutor<>(vertx, tasks, 5);
+        List<Future<Boolean>> taskFutures = concurrentTaskExecutor.start();
+        Future.join(taskFutures).onComplete(context.failing(result -> context.verify(() -> {
+            assertThat(result).isNotNull();
+            taskFutures.forEach(ar -> assertThat(ar.failed()).isTrue());
+            context.completeNow();
+        })));
+    }
+
+    @Test
+    public void testSingleTaskSuccess(VertxTestContext context)
+    {
+        List<Callable<Future<String>>> tasks = Collections.singletonList(getDummySuccessTask("Task executed successfully"));
+        AsyncConcurrentTaskExecutor<String> concurrentTaskExecutor = new AsyncConcurrentTaskExecutor<>(vertx, tasks, 1);
+        List<Future<String>> taskFutures = concurrentTaskExecutor.start();
+        Future.join(taskFutures).onComplete(ar -> context.verify(() -> {
+            assertThat(ar).isNotNull();
+            assertThat(ar.result().list().get(0)).isEqualTo("Task executed successfully");
```

Comment:
I think we need to wrap the future in context.succeeding, otherwise errors will be swallowed 
```suggestion
        Future.join(taskFutures).onComplete(context.succeeding(result -> context.verify(() -> {
            assertThat(result).isNotNull();
            assertThat(result.list().get(0)).isEqualTo("Task executed successfully");
```

Reply:
It failed when there are assertion failures or exceptions. Tried the following and the test did fail
```java
        Future.join(taskFutures).onComplete(ar -> context.verify(() -> {
            assertThat(1).isEqualTo(2);  // Making sure that it fails
            assertThat(ar).isNotNull();
            assertThat(ar.result().list().get(0)).isEqualTo("Task executed successfully");
            context.completeNow();
        }));
```

context.verify is doing this:

```java
  public VertxTestContext verify(ExecutionBlock block) {
    Objects.requireNonNull(block, "The block cannot be null");
    try {
      block.apply();
    } catch (Throwable t) {
      failNow(t);
    }
    return this;
  }
```

As per vertx documentation "_[verify](https://vertx.io/docs/apidocs/io/vertx/junit5/VertxTestContext.html#verify-io.vertx.junit5.VertxTestContext.ExecutionBlock-) to perform assertions, any exception thrown from the code block is considered as a test failure_", I feel context.verify is good enough.